python/pyspark/sql/tests/connect/streaming/test_parity_foreach_batch.py

```diff
@@ -66,6 +66,30 @@ def func(df, _):
             q = df.writeStream.foreachBatch(func).start()
             q.processAllAvailable()
 
+    def test_pickling_deserialization_error(self):
+        class NoUnpickle:
+            def __init__(self, data):
+                self.data = data
+
+            def __reduce__(self):
+                # Serialize only the data attribute
+                return (self.__class__, (self.data,))
```

Comment:
I think here, you should return a function that throws.
```
def __reduce__(self):
  reurn (error_fn, ())

def error_fn:
   raise ...
```

Reply:
Ah yeah, will make the refactor.